python/tests/stress/arcticdb/version_store/test_mem_leaks.py

```diff
@@ -146,13 +160,158 @@ def grow_exp(df_to_grow: pd.DataFrame, num_times_xx2: int):
+    That process if >6-8 times more time efficient. The down side is that you have N**2 repetitions of
+    same original table. But for many usages size is what we need not uniqueness
+    """
+    logger.info("Generating big dataframe")
+    st = time.time()
+    df = get_sample_dataframe(rows)
+    df = grow_exp(df, num_exp_time_growth)
+    logger.info(f"Generation took : {time.time() - st}")
+    return df
+
+#endregion
+
+#region HELPER functions for memray tests
+
+def construct_df_querybuilder_tests(size: int) -> pd.DataFrame:
+    df = get_sample_dataframe(size)
+    df.index = pd.date_range(start="2000-1-1", periods=size, freq="s")
+    df.index.name = "timestamp"
+    return df
+
+
+def query_groupby_with_aggregations() -> QueryBuilder:
+    """
+    groupby composite aggregation query for QueryBuilder memory tests.
+    The query basically will do aggregation of half of dataframe
+    """
+    q = QueryBuilder()
+    return (
+        q[q["bool"]]
+        .groupby("uint8")
+        .agg({"uint32": "mean", 
+              "int32": "sum", 
+              "strings": "count", 
+              "float64": "sum", 
+              "float32": "min", 
+              "int16": "max"})
+    )
+
+
+def query_group_with_aggregations_bigger_result() -> QueryBuilder:
+    """
+    groupby composite aggregation query for QueryBuilder memory tests.
+    The query basically will do aggregation of whole dataframe
+    """
+    q = QueryBuilder()
+    return (
+        q[q["strings"] != "QASDFGH"]
+        .groupby("int16")
+        .agg({"uint32": "mean", 
+              "int32": "sum", 
+              "strings": "count", 
+              "float64": "sum", 
+              "float32": "min", 
+              "int16": "max"})
+    )
+
+
+def query_apply(strng: str) -> QueryBuilder:
+    """
+    Apply query for QueryBuilder memory tests.
+    This version of apply does couple of nested operations
+    with columns that were just added
+    """
+    q = QueryBuilder()
+    q = q[q["strings"] != strng].apply("NEW 1", q["uint32"] * q["int32"] / q["float32"])
+    q = q.apply("NEW 2", q["float32"] + q["float64"] + q["NEW 1"])
+    q = q.apply("NEW 3", q["NEW 2"] - q["NEW 1"] + q["timestamp"])
+    return q
+
+
+def query_resample() -> QueryBuilder:
+    """
+        Resample query for QueryBuilder memory tests
+    """
+    q = QueryBuilder()
+    return q.resample("min").agg(
+        {"int8" : "min",
+        "int16" : "max",
+        "int32" : "first",
+        "int64" : "last",
+        "uint64" : "sum",
+        "float32" : "mean",
+        "float64" : "sum",
+        "strings" : "count",
+        "bool" : "sum"}
+        )
+
+
+def query_row_range(size: int) -> QueryBuilder:
+    """
+    Row range query for QueryBuilder memory tests
+    Pass size of dataframe and it will generate random row range
+    """
+    percentage_rows_returned = 0.57
+    start_percentage = random.uniform(0.01, 1.0 - percentage_rows_returned - 0.01)
```

Comment:
Why are you subtracting 0.01? FYI it is safe to pass row_range(0, size) because the row_range is inclusive of the low end and exclusive of the upper end. https://docs.arcticdb.io/latest/api/library/#arcticdb.version_store.library.Library.read
Subtracting one on row 267 is not necessary